src/providers/mod.rs

```diff
@@ -171,9 +171,16 @@ pub trait MetadataProvider {
     fn hostname(&self) -> Result<Option<String>>;
     fn ssh_keys(&self) -> Result<Vec<PublicKey>>;
     fn networks(&self) -> Result<Vec<network::Interface>>;
-    fn network_devices(&self) -> Result<Vec<network::Device>>;
     fn boot_checkin(&self) -> Result<()>;
 
+    /// Return a list of virtual network devices for this machine.
+    ///
+    /// This is used to setup virtual interfaces, e.g. via [systemd.netdev][netdev]
+    /// configuration fragments.
```

Comment:
OK, I totally missed in the past that Afterburn also took care of this. Are there plans to support NM too? Should the interface we're setting up here be network-provider independent in that case?

Reply:
Yes. This is part of a larger cleanup I'm doing to support NM. That itself is part of larger task to configure RHCOS/FCOS network on ibmcloud classic.

I still don't know how many pieces I can share between NM<->networkd (and how the datastructures would look like), so for the moment I just started by better labeling the things that are clearly networkd-specific.

I guess I'll have a better idea (and additional pass on this) once I get closer to synthesizing NM keyfiles.